src/request/creditcard.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { ArgumentError } from '../errors';
+import CreditCard from './creditcard';
+
+describe('CreditCard()', () => {
+  it('throws an error if avsResult is not valid', () => {
+    expect(() => {
+      const creditcard = new CreditCard({
```

Comment:
I was going to suggest not even assigning the object to `const creditcard` because `creditcard` isn't used elsewhere, but I see that causes TSLint to freak out about `no-unused-expression`.

Reply:
Ah, this could be written as:
```
    expect(() => new CreditCard({
      avsResult: 'foo',
    })).toThrowError(ArgumentError);
```